index.js

```diff
@@ -416,6 +431,32 @@ Primus.readable('forEach', function forEach(fn, done) {
 });
 
 /**
+ * Send a ping packet to all clients to ensure that they are still connected.
+ *
+ * @returns {Primus}
+ * @api private
+ */
+Primus.readable('heartbeat', function heartbeat() {
+  this.forEach(function forEach(spark) {
+    if (!spark.alive) {
```

Comment:
Ok, so two things to do: 

* Set `spark.alive` when a protocol ping comes in, and 
* Add per-spark options:
  * `wantsHeartbeats: true`: Toggles whether or not we should send `primus::ping`.
  * `checkIfZombie: true`: Toggles whether or not we should disconnect if `!spark.alive`.
    - This would help us do a mode that disables heartbeating entirely if needed.